ftw/collectionblock/browser/block_view.py

```diff
@@ -63,6 +69,52 @@ def get_block_info(self):
 
         return info
 
+    def tabular_fields(self):
+        """Returns a list of all metadata fields from the catalog that were
+           selected.
+        """
+        context = aq_inner(self.context)
+        fields = ICollection(context).selectedViewFields()
+        fields = [field[0] for field in fields]
+        return fields
+
+    def tabular_fielddata(self, item, fieldname):
+        value = getattr(item, fieldname, '')
+        if safe_callable(value):
+            value = value()
+        if fieldname in [
```

Comment:
Since we already hat to customise the `def tabular_fielddata()` method in a project where we use this new feature, we could move this `if` to a method `def format_datetime_fielddate(fieldname, value)` so we would only need to customise that method. What do you think?

Reply:
I removed the changes done related to this review again.